src/App.tsx

```diff
@@ -1,18 +1,35 @@
-import React from 'react';
+import React, { ChangeEvent, useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const displayedItems = items.slice(
+    itemsPerPage * (currentPage - 1),
+    itemsPerPage * (currentPage),
+  );
+  const firstItem = displayedItems[0];
+  const lastItem = displayedItems[displayedItems.length - 1];
+
+  const handleItemsAmountPerPage = (event: ChangeEvent<HTMLSelectElement>) => {
+    setItemsPerPage(Number(event.target.value));
+    setCurrentPage(1);
+  };
+
+  const options = [3, 5, 10, 20];
```

Comment:
Move this array outside the component. In this case, this array will not create again and again for each rendering.